pylint/checkers/stdlib.py

```diff
@@ -494,8 +503,12 @@ def visit_call(self, node):
                 if inferred is astroid.Uninferable:
                     continue
                 if inferred.root().name == OPEN_MODULE:
-                    if getattr(node.func, "name", None) in OPEN_FILES:
+                    if getattr(node.func, "name", None) in OPEN_FILES_MODE:
                         self._check_open_mode(node)
+                    if getattr(node.func, "name", None) in {"open"} or getattr(
+                        node.func, "attrname", None
+                    ) in {"open"}:
```

Comment:
```suggestion
                    if (
                        isinstance(node.func, astroid.Name) and node.func.name in ("open",)
                        or isinstance(node.func, astroid.Attribute) and node.func.attrname in ("open",)
                    ):
```
Furthermore, it might be worth considering to replace the `getattr` calls with `isinstance` once.